### PR TITLE
Stop using Function::getBasicBlockList

### PR DESCRIPTION
**Pull request by jayfoad**
It has been made private by upstream LLVM patch
https://reviews.llvm.org/D140162

### CODE REVIEW

File: llpc/lower/llpcSpirvLowerRayQuery.cpp [hunk 1]
```diff
@@ -1612,7 +1612,7 @@ void SpirvLowerRayQuery::createIntersectBvh(Function *func) {
 // Create sample gpu time
 //
 void SpirvLowerRayQuery::createSampleGpuTime(llvm::Function *func) {
-  assert(func->getBasicBlockList().size() == 1);
+  assert(std::distance(func->begin(), func->end()) == 1);
```
Comment (piotrAMD) [inline llpc/lower/llpcSpirvLowerRayQuery.cpp:1615]: Would `func->size()` work here instead?

Comment (jayfoad) [reply llpc/lower/llpcSpirvLowerRayQuery.cpp:1615]: Good point! I did not know that existed.

**Review by piotrAMD — APPROVED** [review]
LGTM - thanks!

**Comment by amdvlk-admin** [comment]
### Test summary for commit b559bdb7547c70ad5685798c04b6fa0e6f3ed12e
<details><summary>CTS tests (Failed: 52/277243)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38797/70350 (55.1%)</li>
<li><details><summary>Failed: 52/70350 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.binding_model.buffer_device_address.set31.depth2.basessbo.crossconvertu2p.nostore.replay.std140.frag
Stack trace: Flake

FAILURE: dEQP-VK.binding_model.shader_access.primary_cmd_buf.with_template.storage_image.vertex.multiple_discontiguous_descriptors.2d_base_mip
Stack trace: Flake

FAILURE: dEQP-VK.clipping.user_defined.clip_distance_dynamic_index.vert.3_fragmentshader_read
Stack trace: Flake

FAILURE: dEQP-VK.compute.workgroup_memory_explicit_layout.zero.u8vec4_array_to_double_float16_t_array_4
Stack trace: Flake

FAILURE: dEQP-VK.compute.workgroup_memory_explicit_layout.zero.uvec4_array_to_float_dvec4_array_4
Stack trace: Flake

FAILURE: dEQP-VK.glsl.440.linkage.varying.component.vert_out_frag_in.vec3.as_float_vec2
Stack trace: Flake

FAILURE: dEQP-VK.glsl.builtin_var.fragcoord_msaa_input.8_bit
Stack trace: Flake

FAILURE: dEQP-VK.glsl.conversions.vector_to_vector.bvec4_to_uvec2_fragment
Stack trace: Flake

FAILURE: dEQP-VK.glsl.operator.binary_operator.add_assign_result.highp_vec3_vertex
Stack trace: Flake

FAILURE: dEQP-VK.glsl.texture_functions.query.texturequerylod.sampler1darray_float_fragment
Stack trace: Flake

FAILURE: dEQP-VK.glsl.texture_functions.textureprojlodoffset.sampler1d_vec4_fixed_vertex
Stack trace: Flake

FAILURE: dEQP-VK.graphicsfuzz.cov-folding-rules-dot-no-extract
Stack trace: Flake

FAILURE: dEQP-VK.image.image_size.1d_array.readonly_writeonly_12x34
Stack trace: Flake

FAILURE: dEQP-VK.image.mismatched_formats.image_write.r8g8b8a8_unorm_with_rgba8
Stack trace: Flake

FAILURE: dEQP-VK.image.nontemporal_operand.r16g16_sint
Stack trace: Flake

FAILURE: dEQP-VK.memory_model.shared.16bit.vector_types.1
Stack trace: Flake

FAILURE: dEQP-VK.multiview.index.fragment_shader.15_15_15_15
Stack trace: Flake

FAILURE: dEQP-VK.query_pool.statistics_query.geometry_shader_primitives.secondary_inherited.64bits_line_list_with_adjacency
Stack trace: Flake

FAILURE: dEQP-VK.ray_query.acceleration_structures.flags.fragment_shader.gpu_built.triangles_aop.identical_instances.nopadding.fastbuild_update_0_0_topgeneric
Stack trace: Flake

FAILURE: dEQP-VK.ray_query.traversal_control.chit_shader.skip_intersection.aabbs
Stack trace: Flake
...
```
</p></details></li>

<li>Not Supported: 31501/70350 (44.8%)</li><li>Warnings: 0/70350 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 39099/70350 (55.6%)</li>
<li>Failed: 0/70350 (0.0%)</li>
<li>Not Supported: 31251/70350 (44.4%)</li><li>Warnings: 0/70350 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35944/70352 (51.1%)</li>
<li>Failed: 0/70352 (0.0%)</li>
<li>Not Supported: 34408/70352 (48.9%)</li><li>Warnings: 0/70352 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx9</summary><ul>

<li>Passed: 33663/66191 (50.9%)</li>
<li>Failed: 0/66191 (0.0%)</li>
<li>Not Supported: 32528/66191 (49.1%)</li><li>Warnings: 0/66191 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit a263751ef4f409a1649caee0916f24904cd92f67
<details><summary>CTS tests (Failed: 0/277243)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38849/70350 (55.2%)</li>
<li>Failed: 0/70350 (0.0%)</li>
<li>Not Supported: 31501/70350 (44.8%)</li><li>Warnings: 0/70350 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 39099/70350 (55.6%)</li>
<li>Failed: 0/70350 (0.0%)</li>
<li>Not Supported: 31251/70350 (44.4%)</li><li>Warnings: 0/70350 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35944/70352 (51.1%)</li>
<li>Failed: 0/70352 (0.0%)</li>
<li>Not Supported: 34408/70352 (48.9%)</li><li>Warnings: 0/70352 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx9</summary><ul>

<li>Passed: 33663/66191 (50.9%)</li>
<li>Failed: 0/66191 (0.0%)</li>
<li>Not Supported: 32528/66191 (49.1%)</li><li>Warnings: 0/66191 (0.0%)</li>
</ul></details>


</ul></details>

**Review by nhaehnle — APPROVED** [review]

**Comment by amdvlk-admin** [comment]
### Test summary for commit 2a23f6e516935ca571bd955ef8d24e145c9ca8b4
<details><summary>CTS tests (Failed: 0/277243)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38849/70350 (55.2%)</li>
<li>Failed: 0/70350 (0.0%)</li>
<li>Not Supported: 31501/70350 (44.8%)</li><li>Warnings: 0/70350 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 39099/70350 (55.6%)</li>
<li>Failed: 0/70350 (0.0%)</li>
<li>Not Supported: 31251/70350 (44.4%)</li><li>Warnings: 0/70350 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35944/70352 (51.1%)</li>
<li>Failed: 0/70352 (0.0%)</li>
<li>Not Supported: 34408/70352 (48.9%)</li><li>Warnings: 0/70352 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx9</summary><ul>

<li>Passed: 33663/66191 (50.9%)</li>
<li>Failed: 0/66191 (0.0%)</li>
<li>Not Supported: 32528/66191 (49.1%)</li><li>Warnings: 0/66191 (0.0%)</li>
</ul></details>


</ul></details>

**Review by dstutt — APPROVED** [review]

**Pull request merged**